handlers.go

```diff
@@ -262,30 +262,26 @@ func (dht *IpfsDHT) handleFindPeer(ctx context.Context, from peer.ID, pmes *pb.M
 
 	// if looking for self... special case where we send it on CloserPeers.
 	targetPid := peer.ID(pmes.GetKey())
-	if targetPid == dht.self {
-		closest = []peer.ID{dht.self}
-	} else {
-		closest = dht.betterPeersToQuery(pmes, from, dht.bucketSize)
-
-		// Never tell a peer about itself.
-		if targetPid != from {
-			// Add the target peer to the set of closest peers if
-			// not already present in our routing table.
-			//
-			// Later, when we lookup known addresses for all peers
-			// in this set, we'll prune this peer if we don't
-			// _actually_ know where it is.
-			found := false
-			for _, p := range closest {
-				if targetPid == p {
-					found = true
-					break
-				}
-			}
-			if !found {
-				closest = append(closest, targetPid)
+	closest = dht.betterPeersToQuery(pmes, from, dht.bucketSize)
+
+	// Never tell a peer about itself.
+	if targetPid != from {
+		// Add the target peer to the set of closest peers if
+		// not already present in our routing table.
+		//
+		// Later, when we lookup known addresses for all peers
+		// in this set, we'll prune this peer if we don't
+		// _actually_ know where it is.
+		found := false
+		for _, p := range closest {
+			if targetPid == p {
+				found = true
+				break
 			}
 		}
+		if !found {
+			closest = append(closest, targetPid)
+		}
```

Comment:
Why do we do this? If the peerID isn't found, should we return it in findPeer?

Reply:
[`dht.betterPeersToQuery`](https://github.com/libp2p/go-libp2p-kad-dht/blob/master/dht.go#L765-L769) never includes itself (nor `from`) in the results.

Currently if `A` sends `FIND_NODE(B)` to `B`, `B` will reply `[B]`, and we now want it to return `[B, C, D, ...]`. In this case `targetPid` is our own peer id, it wasn't returned by `dht.betterPeersToQuery`, so we need to add it at this step. 

We will be able to get rid of this when the response to `FIND_NODE(self)` will be not include `self`.

Reply:
But if the target node doesnt exist we will include it in the output, right?

Reply:
Peers without addresses are excluded from the response

https://github.com/libp2p/go-libp2p-kad-dht/blob/7d6120fb0d6706c4ca76f5b3228747e0749670f5/handlers.go#L292-L301